src/Uart/Uart.c

```diff
@@ -314,24 +319,38 @@ handleTxInterrupt(Uart *const uart)
 	}
 }
 
-void
-Uart_handleInterrupt(Uart *const uart)
+static inline bool
+Uart_hasAnyErrorOccured(Uart_ErrorFlags* const errFlags)
 {
-	uint32_t status = uart->reg->sr;
+	return (errFlags->hasFramingErrorOccurred || errFlags->hasOverrunOccurred || errFlags->hasParityErrorOccurred
+            || errFlags->hasRxFifoFullErrorOccurred);
+}
+
+bool
+Uart_handleInterrupt(Uart *const uart, int * errCode)
+{
+	bool retValue = true;
+	Uart_ErrorFlags errorFlags = { false, false, false, false };
+
+	uint32_t status = uart->reg->sr & uart->reg->imr;
 	uart->reg->cr = UART_CR_RSTSTA_MASK;
 	if ((status & UART_SR_RXRDY_MASK) != 0u)
-		handleRxInterrupt(uart);
+	{
+	    retValue &= handleRxInterrupt(uart, errCode);
+	    if(*errCode == Uart_ErrorCodes_Rx_Fifo_Full) {
```

Comment:
returnError(errCode, Uart_ErrorCodes_Rx_Fifo_Full); is null pointer safe if I recall. *errCode is not. But in the runtime you have a call: Uart_handleInterrupt(&halUart.uart, NULL); Please make it null pointer safe, as this is the assumed convention in our code.

Reply:
ACK